crates/vm/src/eval.rs

```diff
@@ -329,6 +347,37 @@ pub mod tests {
         assert_eq!(result.as_without_table_name(), input.as_without_table_name(), "Project");
     }
 
+    #[test]
+    fn test_semijoin() {
+        let p = &mut Program::new(AuthCtx::for_testing());
+        let table = MemTable::from_value(scalar(1));
+        let field = table.get_field_pos(0).unwrap().clone();
+
+        let mut sources = SourceSet::default();
+        let source_expr = sources.add_mem_table(table.clone());
+        let second_source_expr = sources.add_mem_table(table);
+
+        let q = query(source_expr).with_join_inner(second_source_expr, field.clone(), field, true);
```

Comment:
Again, I think I would prefer a higher level test that goes from sql to result set. You can use `RelationalDB::create_table_for_test` along with `sql::execute::run`.